src/main/java/uk/gov/pay/ledger/transaction/model/PaymentFactory.java

```diff
@@ -0,0 +1,84 @@
+package uk.gov.pay.ledger.transaction.model;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.ledger.transaction.entity.TransactionEntity;
+import uk.gov.pay.ledger.transaction.search.model.RefundSummary;
+import uk.gov.pay.ledger.transaction.search.model.SettlementSummary;
+import uk.gov.pay.ledger.transaction.state.TransactionState;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.lang.String.format;
+
+public class PaymentFactory {
+
+    private static Logger LOGGER = LoggerFactory.getLogger(PaymentFactory.class);
+    private ObjectMapper objectMapper;
+
+    @Inject
+    public PaymentFactory(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    public Payment fromTransactionEntity(TransactionEntity entity) {
+        try {
+            JsonNode transactionDetails = objectMapper.readTree(Optional.ofNullable(entity.getTransactionDetails()).orElse("{}"));
+            Address billingAddress = Address.from(
```

Comment:
This could be a constructor? It seems to be directly passing in params to make an Address object which doesn't seem like the right fit for a `from` pattern

Reply:
It can't be constructor. The from method might return `null` and constructors can't return `null`.
https://github.com/alphagov/pay-ledger/blob/master/src/main/java/uk/gov/pay/ledger/transaction/model/Address.java#L30-L41